crates/engine/tree/src/tree/mod.rs

```diff
@@ -2270,9 +2350,16 @@ where
         let block_hash = block.hash();
         let sealed_block = Arc::new(block.clone_sealed_block());
 
-        let persistence_not_in_progress = !self.persistence_state.in_progress();
+        // We only run the parallel state root if we are currently persisting blocks that are all
+        // ancestors of the one we are executing. If we're committing ancestor blocks, then: any
+        // trie updates being committed are a subset of the in-memory trie updates collected before
+        // fetching reverts. So any diff in reverts (pre vs post commit) is already covered by the
+        // in-memory trie updates we collect in `compute_state_root_parallel`.
+        //
+        // See https://github.com/paradigmxyz/reth/issues/12688 for more details
+        let is_descendant_block = self.is_descendant_of_persisting_blocks(block.header());
```

Comment:
`is_descendant_block` when used later might be not clear enough? maybe same as the method name, `is_descendant_of_persisting_blocks`,  would be better, wdyt?

Reply:
Yeah that's a better name

Reply:
renamed